src/astgen.zig

```diff
@@ -2867,26 +2877,7 @@ fn identifier(
                 ),
                 error.InvalidCharacter => break :integer,
             };
-            const val = switch (bit_count) {
-                8 => if (is_signed) Value.initTag(.i8_type) else Value.initTag(.u8_type),
-                16 => if (is_signed) Value.initTag(.i16_type) else Value.initTag(.u16_type),
-                32 => if (is_signed) Value.initTag(.i32_type) else Value.initTag(.u32_type),
-                64 => if (is_signed) Value.initTag(.i64_type) else Value.initTag(.u64_type),
-                else => {
-                    return rvalue(mod, scope, rl, try addZIRInstConst(mod, scope, src, .{
-                        .ty = Type.initTag(.type),
-                        .val = try Value.Tag.int_type.create(scope.arena(), .{
-                            .signed = is_signed,
-                            .bits = bit_count,
-                        }),
-                    }));
-                },
-            };
-            const result = try addZIRInstConst(mod, scope, src, .{
-                .ty = Type.initTag(.type),
-                .val = val,
-            });
-            return rvalue(mod, scope, rl, result);
+            return rvalue(mod, scope, rl, try gz.addBin(.int_type, @boolToInt(is_signed), bit_count), src);
```

Comment:
This has the same type safety issue (src is the wrong type here) 

Also in the case of common integer types, we want to avoid allocating an instruction entirely, by taking advantage of the `zir.Const` enum.